### ISSUE TITLE
[BUG] Addon not usable on IOS devices due to missing meanings

### ISSUE THREAD

**Issue by Epicrex** [migaku-official/Migaku-Kanji-Addon#171]
### The Bug
I'm not seeing the english meaning of the Kanji on IOS. I tried this deck on my android phone as well as on my windows pc and there it works as it should.

### To Reproduce
1. Open any Mikagu Kanji generated deck
2. Look at the english meaning

### Following device of mine is having the issue
 - iPhone 12 pro, IOS 16
 - Anki Version 2.0.87 (20087.1) (Newest anki version on IOS)

### Screenshots
IOS 16 example (as you can see the meanings are missing):
<img src="https://user-images.githubusercontent.com/85735034/210428082-f212006f-22f3-42b0-8a92-6461713cd23c.png" width ="250">

Windows example (here it's working):
<img src="https://user-images.githubusercontent.com/85735034/210428698-5b97f039-8414-4c07-8c21-fd148ecb4f47.png" width ="400">

**Comment by Epicrex** [comment]
Any updates on if this is being worked on?

**Comment by Epicrex** [comment]
This makes the Migaku-Kanji-Addon unusable for 50% of users why isn’t this top priority. It’s been 4 months since I brought this issue to light and still no peep regarding it.

**Comment by saxoncameron** [comment]
Hey @Epicrex, so sorry you're having this issue, and apologies for the radio silence on our end. I understand your frustration, this seems like a nasty issue that affects core usage of this particular piece of software.

Unfortunately I can't personally replicate this issue, and regretfully Kanji God in general is not high on our long list on priorities as it's part of our legacy software set. Migaku has a very wide and ambitious suite of software with many features. Internally we're working hard to transition to next-generation software that better consolidates all of our existing products and features in a way that's both nicer to use and easier to maintain. Until we reach that goal, there's unfortunately a limbo period where updates, fixes, and general maintenance of our existing software isn't something we can sink days and days into lest it delay any of our already long-anticipated updates.

That's not to say we won't fix this issue, just wanted to share some reasons why you might feel as though you've been left in the dark on this one. To further triage this issue, I've reached out to the wider community to see if anyone else is experiencing this issue: https://discord.com/channels/752293144917180496/846925957302714388/1094416551706370189

**Comment by MichaelPetre** [comment]
I've been studying on ipad (9th gen) for the last 12 months or so. The issue is limited to a couple of cards and 目 happens to be one of them. 

MobileAnki 20090.2 on iOS 16.3.1
Another card, complete
![](https://cdn.discordapp.com/attachments/846925957302714388/1094514670112608367/IMG_0012.png)

目 missing keyword, meaning, vocab and stories:
![](https://user-images.githubusercontent.com/85932194/230761241-383a8409-5935-4412-99cc-65b44bacae8a.jpg)

On PC:
![](https://cdn.discordapp.com/attachments/846925957302714388/1094528100743794748/image.png)

![](https://cdn.discordapp.com/attachments/846925957302714388/1094528190409621504/image.png)

**Comment by Epicrex** [comment]
@MichaelPetre thanks for pointing this out, 3 kanji have this problem but after that most show the words.

When I'm already talking about Kanji good: I seem to be getting no new Kanji put into this kanji god controlled deck (it's always the same 20 Kanji) even though I linked it my core 2/6k optimized deck, from which it should be pulling kanji preemptively. And I'm regularly using this 2/6k optimized deck. Do you maybe know why I'm having this issue

**Comment by Epicrex** [comment]
@saxoncameron there are many options out there for learning kanji, but I want to use Migakus tools, because the presentation of these kanji cards is to beautiful and contains everything one might want down to stroke order. So I find it quite sad that the only way to get these kanji cards is through a quite complex setup in Anki. I would be happy to pay for a standalone app from migaku for learning kanji, although I find subscription models quite off putting.

Having the kanji god addon in Anki however, also comes with it's perks. Being able to learn kanji at the point when one needs them for the core 2/6k deck. But Maybe consider making the setup process easier and releasing a concise guide on setting it up. I had to sift through a couple of quite hard to find youtube videos to set it up correctly. And I'm quite tech savvy, I don't even want to know what a nightmare this is to setup as someone who isn't that familiar. Added to that I don't even think I set it up correctly, because of some other issues I had at that point.

**Comment by saxoncameron** [comment]
@Epicrex Kanji God was also my favourite Migaku product when I was also a poweruser (before I joined Migaku), and it's usefulness is what motivated me to contribute UI improvement (the card UI) for free since the repo is open source.

KG is super powerful and flexible and feature rich, but yeah, not easy or intuitive to set up, and is also dependent on Anki on desktop. So it works for powerusers but even then can be a little frustrating. The same is also true of our browser extension and like I said, we are working hard to consolidate and improve all of these things to make them easy to setup and use (and not just for powerusers). In the medium term future we will have a kanji dictionary and Kanji-God like features inbuilt into our own SRS which will work cross-platform and device. But it takes time to make quality software...!

**Comment by saxoncameron** [comment]
Thanks Michael, that's useful, and might be a lead. If we compare what's different about those cards we might find the root of the problem 🤔

**Comment by saxoncameron** [comment]
> @MichaelPetre thanks for pointing this out, 3 kanji have this problem but after that most show the words.
> 
> When I'm already talking about Kanji good: I seem to be getting no new Kanji put into this kanji god controlled deck (it's always the same 20 Kanji) even though I linked it my core 2/6k optimized deck, from which it should be pulling kanji preemptively. And I'm regularly using this 2/6k optimized deck. Do you maybe know why I'm having this issue

Maybe open a separate issue for this, or reach-out/search on the server for this one. _Probably_ as setup issue that might come down to how things are configured in the many KG menu tabs

**Comment by Epicrex** [comment]
@saxoncameron after the kanji god controlled deck has been set up correctly on the Anki desktop version. Is it able to add new kanji cards to the kanji god controlled deck based on upcoming cards in the linked deck, even when I'm only using Anki on my phone?

Because the issue could be that Kanji god isn't able to add new cards when on mobile. And all the cards I can see in the kanji god controlled deck are those that where added from the initial creation of that deck.

**Comment by saxoncameron** [comment]
Yes, unfortunate one major limitation is its dependence on desktop Anki due to there being no addon support on mobile

**Comment by MichaelPetre** [comment]
> Because the issue could be that Kanji god isn't able to add new cards when on mobile. And all the cards I can see in the kanji god controlled deck are those that where added from the initial creation of that deck.

Yes, Kanji God only runs on the desktop, so the new cards will not get created on the ipad. The workaround is to "force" the creation of the kanji cards by entering and exiting the Kanji Settings window on the desktop (after syncing back to desktop, of course).

My daily anki workflow due to the iOS limitations:

-Study on the ipad during the commute
-Sync back from iPad to ankiweb once I have wifi
-Sync from ankiweb to desktop once back home
-Optional: mining from shows/novels 
-Morphman recalc
-Open and close Kanji Settings on the desktop (mandatory, this will create the new kanji cards from your vocab deck studied on ipad)
-Sync back from desktop to ankiweb
-Sync from ankiweb to ipad

That way, every morning my cards in Anki mobile are organized for the shortest path to my target and contain all the required new kanji. It only took a couple of days to get used to that workflow.

**Comment by Epicrex** [comment]
@MichaelPetre thanks for telling me about your workflow. 

I've really noticed, that learning words without knowing what the kanji means makes it so much harder for me to remember the word, so the system of learning the kanji/ kanji radicals before learning Japanese words that use them is invaluable. 

I really hope Migaku will soon be releasing their tool which incapsulates this but until then I guess this solution will do.

**Comment by Epicrex** [comment]
@saxoncameron  Think of when you are driving, doing a workout or anything else which preoccupies your eyes, wouldn’t it be great to be able to use that time to learn Japanese.

So my suggestion is to have some mode where cards one is currently learning are read out aloud. 

More details: It could read the English word and then repeat the Japanese word a couple of times. It could also say the english word and the pause so the user can try to come up with the world on their own and then say the Japanese word.

What effort this would entail on migakus end: Currently their is no sound for English words in Anki an probably also not in Migakus upcoming software, so that would have to be added. But I don’t think that should be to much work because their are probably already preexisting project where allot of english words are pronounced or one could use AI to pronounce these english words.

Also a cool feature to add to the kanji cards, would be to show the vocab one has learned already that uses that kanji.

On another note, I’m quite excited for this Migaku software/ app and am happy to help the team with input on anything. I have a fair amount of experience with UI/ UX design.
